mbf_abstract_nav/src/controller_action.cpp

```diff
@@ -131,160 +133,164 @@ void ControllerAction::run(GoalHandle &goal_handle, AbstractControllerExecution
 
   while (controller_active && ros::ok())
   {
-    if (!robot_info_.getRobotPose(robot_pose))
-    {
-      controller_active = false;
-      fillExePathResult(robot_pose, goal_pose,
-                        mbf_msgs::ExePathResult::TF_ERROR, "Could not get the robot pose!", result);
-      concurrency_slots_[slot].goal_handle.setAborted(result, result.message);
-      ROS_ERROR_STREAM_NAMED(name_, result.message << " Canceling the action call.");
-      break;
-    }
-
-    if (first_cycle)
     {
```

Comment:
Use a method `getGoalHandle` where the slot is locked inside, returning a copy of the goal handle, or use manual mtx.lock(), mtx.unlock() here. This avoids this code remapping and can safely be used everywhere.